src/rapids_dependency_file_generator/rapids_dependency_file_generator.py

```diff
@@ -35,9 +35,12 @@ def delete_existing_files(root="."):
         for fn in filter(
             lambda fn: fn.endswith(".txt") or fn.endswith(".yaml"), filenames
         ):
-            with open(file_path := os.path.join(dirpath, fn)) as f:
-                if HEADER in f.read():
-                    os.remove(file_path)
+            try:
+                with open(file_path := os.path.join(dirpath, fn)) as f:
+                    if HEADER in f.read():
+                        os.remove(file_path)
+            except UnicodeDecodeError:
+                pass
```

Comment:
Let's swap the scope of the `try` and `with` here, since it's the `f.read()` that is problematic. I spoke with @vyasr offline and we agreed this is a good solution.
```suggestion
            with open(file_path := os.path.join(dirpath, fn)) as f:
                try:
                    if HEADER in f.read():
                        os.remove(file_path)
                except UnicodeDecodeError:
                    pass
```